lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -213,7 +213,24 @@ protected function connectTo($connectionName)
         $user = isset($connectionParams['user']) ? $connectionParams['user'] : null;
         $password = isset($connectionParams['password']) ? $connectionParams['password'] : null;
 
-        return $this->_driver->connect($connectionParams, $user, $password, $driverOptions);
+        if ($connectionName == 'slave') {
+            while(true) {
```

Comment:
If the connection in line 219 is successfull it exits the loop by the return. If it fails it leaves the loop by throwing an exception as soon as no slaves are left. Did i miss any case? Or do you just prefer to not have a ``while(true)`` in the code to preserve faults in futere changes?